Add jstack (8,11,12) and update jps (8,11)

SueChaplain opened the pull request:
- New tools topic for jstack
- New section for jstack in the release topic
- Updated to Tools "Switching to OpenJ9" to describe differences to
  HotSpot jstack

- Add jps to release topic for 8 & 11 (12 was in 0.13.0)
- Remove icons from jps content that specify Java 12 only.

Closes: #225

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

SueChaplain commented:
Jenkins doc stage

pshipton commented:
Note there is a merge conflict.

SueChaplain commented:
Jenkins doc stage

pshipton commented on docs/tool_jstack.md:
```diff
@@ -0,0 +1,50 @@
+
+- `-p`: Prints the system and agent properties of the process
+- `-J`: supplies arguments to the Java VM that is running the `jstack` command. You can use multiple `-J` options, for example: `jstack -J-Xmx10m -J-Dcom.ibm.tools.attach.enable=yes`
+- `-l`: prints the application package name
```
According to the jstack help output, these are the only options supported
```
				+ "    -p: print the target's system and agent properties%n"
				+ "    -l: Long format. Print the thread's ownable synchronizers%n"
				+ "    -J: supply arguments to the Java VM running jps%n"
```

SueChaplain commented:
@pdbain-ibm @pshipton 

Please can this be reviewed your time today so that any changes I need to make can happen tomorrow, after which I am on vacation for a week.

I'm unsure about the list of options, because the issue says *The usual suspects: -J, -h, like jps.*

I've also added some references to Linux in there where I think options/syntax is not possible on Windows platforms. Might need some tweaks.

I've staged the doc but the latest updates are still building, so you'll need to wait for ghpages to refresh before checking: https://eclipse.github.io/openj9-docs-staging/246/tool_jstack/

I changed the title of the topic to "Java stack tool", because it doesn't look good in the table of contents, so when you see that change you'll know its updated.

pdbain-ibm commented:
@SueChaplain Thanks for working on this!

> I think options/syntax is not possible on Windows platforms

You could say "user __running a Bourne or equivalent shell__ to..."

Please drop these options:
```
-m: prints the application arguments
-v: prints the Java VM arguments, including those that are produced automatically
```
and add the same text for `-h` as `jps`.

`If this option is used on its own,`: I am not sure what you mean by this.  I don't believe it is necessary.

> I changed the title of the topic to "Java stack tool", because it doesn't look good in the table of contents, so when you see that change you'll know its updated.

Could we include the actual tool name table of contents, e.g. `Java stack tool (jstack)` or `jstack: Java stack tool`, or other variant of these (same for `jps`). This makes it clear we have these tools and helps search engines find them.

For reference:
https://docs.oracle.com/javase/7/docs/technotes/tools/share/jstack.html is titled `jstack - Stack Trace` and https://docs.oracle.com/javase/7/docs/technotes/tools/share/jps.html is titled `jps - Java Virtual Machine Process Status Tool`.  For reference also: https://docs.oracle.com/javase/7/docs/technotes/tools/

DanHeidinga commented:
@SueChaplain Oracle makes it very clear that the j* tools are 
> are unsupported and experimental, and should be used with that in mind. They may not be available in future JDK versions.

We should include similar restrictions in openj9 documentation for the these tools

pshipton commented on docs/tool_jstack.md:
```diff
@@ -0,0 +1,50 @@
+- `-v`: prints the Java VM arguments, including those that are produced automatically
+
+
+<i class="fa fa-exclamation-triangle" aria-hidden="true"></i> **Restrictions:**
```
> We should include similar restrictions in openj9 documentation for the these tools

@DanHeidinga is this what you are looking for, or does it need to be clarified or stated in other places as well?

DanHeidinga replied:
Yup, that's exactly what I was looking for.  Missed it on first read

SueChaplain commented:
@pdbain-ibm 

Some confusion over options still:

>add the same text for `-h` as jps

There is no `-h` option for `jps`. I assume you mean that you have added a `-h` to print help information, to make this compatible with `jstack`.  I added this.

>Please drop these options: (-m -v)

Done.

So, with these corrected we still have some discrepancies:

| HotSpot option | OpenJ9 option |
|-------------------|-----------------|
| -J | -J |
| -l | -l |
| -h | -h |
| -help | - |
| -F| not supported |
| -m| native info not supported|
|- | -p |
| -| -q |

You  gave us the `-p` option in your issue, but not `-q`, which got copied from `jps` because of the *all the usual suspects* comment. I assume we remove `-q` here (which I have done in my latest update)? Please confirm. Please also confirm that `-p` is correct.

Forgive me for not looking directly at the Oracle documentation when writing this up. We deliberately avoid doing so because of copyright infringement. It is therefore important that we have all the information directly from you.

>If this option is used on its own,: I am not sure what you mean by this. I don't believe it is necessary.

I meant if it is used as `jstack <pid*>` without adding any `<options>`. I removed the words anyway as the sentence is correct without them.

>Could we include the actual tool name table of contents

I played around with this a bit yesterday. I do agree with you. On their own they look odd. Although we try to avoid long TOC titles, we have longer ones, so I've added in *Full name (cmdname)* for `jstack` and `jps`. Hope that works for you.

SueChaplain commented:
Jenkins doc stage

SueChaplain commented on docs/tool_jstack.md:
```diff
@@ -0,0 +1,50 @@
+
+    jstack <options>* <pid>*
+
+Where `<pid>*` is a list of process IDs. If none are supplied, the process IDs are read from `stdin`, which allows a user to query all processes via `jps -q | jstack`. IDs of inactive processes are silently ignored. If this option is used on its own, the output contains Java stacks and thread information of the specified processes (equivalent to the information provided in `java.lang.managmenet.T ...
```
Fixed

SueChaplain commented:
Jenkins doc stage

SueChaplain commented:
Jenkins doc stage

pdbain-ibm commented:
@SueChaplain 
Thanks for the updates. Reviewed and approved.

SueChaplain commented:
@pshipton - if you are also ok with the updates, please go ahead and merge.

SueChaplain commented:
@pshipton I'm out until after MS1, so if there are any other changes needed these will have to be done on top of my merge, or in a separate PR. Thanks!
